### PR TITLE
mapped ga3 client side events to ga4

### PR DESCRIPTION
**Pull request by fstefan7**

### CODE REVIEW

**Comment by fstefan7** [comment]
[Client side parameter mapping between GA3 and GA4](https://trello.com/c/aM7DNc1g/4279-client-side-parameter-mapping-between-ga3-and-ga4)

File: src/gaTracker/helpers.ts [hunk 1]
```diff
@@ -252,3 +205,106 @@ const addUTMMediumIfMissing = (url: string) => {
...
+			items: convertProductsToGa4Format(products),
+			item_list_name: listName,
+			item_list_id: listName,
+		});
+	} else {
+		const gaProducts = products.map(product => filterGAProductFields(product));
+		gtag('event', 'view_item_list', {
+			event_category,
+			items: gaProducts,
+			send_to: LittledataLayer.webPropertyID,
```
Comment (edwardupton) [inline src/gaTracker/helpers.ts:222]: I think send_to is still needed with GA4: https://developers.google.com/analytics/devguides/collection/ga4/translate-events#if_you_currently_use_gtagjs

File: src/gaTracker/helpers.ts [hunk 2]
```diff
@@ -252,3 +205,106 @@ const addUTMMediumIfMissing = (url: string) => {
 
 	return url.replace(sourceTag, utmTags);
 };
+
+function sendViewItemListEvent(products: Impression[]): void {
+	if (isGa4Format()) {
+		const listName = (products && products.length && products[0].list_name) || '';
+		gtag('event', 'view_item_list', {
```
Comment (edwardupton) [inline src/gaTracker/helpers.ts:212]: In each case it is the event properties which are changing, not the name. So I would refactor to have a `getViewItemListProperties` function, which decides on the properties depending on `isGa4Format()`

File: src/gaTracker/helpers.ts (hunk 1 above)
Comment (edwardupton) [reply src/gaTracker/helpers.ts:222]: We should allow sending to BOTH GA3 and GA4. So if both webPropertyID and measurementID are in the data layer, send the same event to both.

File: src/gaTracker/helpers.ts [hunk 3]
```diff
@@ -252,3 +205,106 @@ const addUTMMediumIfMissing = (url: string) => {
...
+			event_callback() {
+				window.clearTimeout(self.timeout);
+				document.location.href = self.href;
+			},
+		});
+	}
+}
+
+function isGa4Format(): boolean {
+	return LittledataLayer.webPropertyID && LittledataLayer.webPropertyID.includes('G-');
```
Comment (edwardupton) [inline src/gaTracker/helpers.ts:294]: No, it should be called `LittledataLayer.measurementID` for consistency. Don't rely on parsing the string

**Review by edwardupton — CHANGES_REQUESTED** [review]

**Review by edwardupton — APPROVED** [review]

**Review by edwardupton — APPROVED** [review]
convertProductsToGa4Format could really do with unit tests, but i'll approve this to add later

**Comment by fstefan7** [comment]
@edwardupton I'll add a card for adding unit tests for the mapping

**Pull request merged**